libpod/lock/shm/shm_lock.go

```diff
@@ -0,0 +1,188 @@
+package shm
+
+// #cgo LDFLAGS: -lrt -lpthread
+// #include "shm_lock.h"
+// const uint32_t bitmap_size_c = BITMAP_SIZE;
+import "C"
+
+import (
+	"syscall"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	bitmapSize uint32 = uint32(C.bitmap_size_c)
+)
+
+// SHMLocks is a struct enabling POSIX semaphore locking in a shared memory
+// segment.
+type SHMLocks struct { // nolint
+	lockStruct *C.shm_struct_t
```

Comment:
It would be nice for either this `struct`, or the owning `SHMLockManager`, to be a private type, so that the callers only get an interface and can’t accidentally make a copy of the C pointer.